### PR TITLE
IR Sets

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

**Comment by danking** [comment]
@patrick-schultz I'm moving the base over to the stack, but everything should turn out OK.

File: src/main/scala/is/hail/expr/ir/Compile.scala [hunk 1]
```diff
@@ -307,15 +307,114 @@ object Compile {
...
       case GetFieldMissingness(o, name) =>
         val t = o.typ.asInstanceOf[TStruct]
         val fieldIdx = t.fieldIdx(name)
         val (doo, mo, vo) = compile(o)
         present(Code(doo, mo || !t.isFieldDefined(region, coerce[Long](vo), fieldIdx)))
 
+      case MakeSet(args, elementType) =>
+        val srvb = new StagedRegionValueBuilder(fb, TArray(elementType))
+        println(elementType)
+        println(args.toSeq)
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/expr/ir/Compile.scala]: debugging code?

File: src/main/scala/is/hail/annotations/InplaceSort.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package is.hail.annotations
+
+import is.hail.utils._
+import is.hail.expr.ir.TypeToTypeInfo
+import is.hail.expr._
+import is.hail.asm4s._
+
+import org.objectweb.asm.Opcodes._
+import org.objectweb.asm.tree._
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/annotations/InplaceSort.scala]: Unused imports

File: src/main/scala/is/hail/expr/ir/Compile.scala [hunk 3]
```diff
@@ -307,15 +307,114 @@ object Compile {
...
+        val addElement = srvb.addIRIntermediate(elementType)
+        val mvargs = args.map(compile(_))
+        val sortedArray = Code(
+          srvb.start(args.length, init = true),
+          Code(mvargs.map { case (dov, m, v) =>
+            Code(dov, m.mux(srvb.setMissing(), addElement(v)), srvb.advance())
+          }: _*),
+          InplaceSort(region, fb, srvb.offset, TArray(elementType)),
+          srvb.offset)
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/expr/ir/Compile.scala]: Should this be doing duplicate removal?

**Comment by patrick-schultz** [comment]
Oops, didn't notice you moved it.

**Pull request closed without merging**